### PR TITLE
Add users UI for admins

### PR DESCRIPTION
**Pull request by anbraten**
# Changes
- [x] add UI to create, view, update (currently nothing to update) and remove users
- [x] remove active flag as it wasn't used

image

### CODE REVIEW

File: server/api/users.go [hunk 1]
```diff
@@ -58,7 +58,8 @@ func PatchUser(c *gin.Context) {
 		c.AbortWithStatus(http.StatusNotFound)
 		return
 	}
-	user.Active = in.Active
+
+	// TODO: currently nothing is updatable
```
Comment (6543) [inline server/api/users.go:62]: in should be used and only the  `in.id = user.id` to make sure id is not changed ...
or if we allow partial changes... make a check if a field in `in` has a value and replace it here in `user` ...

PS: securety: we make sure only allowe user can change protected fields like Admin etc ...

Comment (anbraten) [reply server/api/users.go:62]: Sure. Currently users are updated by the oauth systems, so in 99% there currently is no need to update an user at all.

File: server/model/user.go [hunk 2]
```diff
@@ -56,9 +56,6 @@ type User struct {
 	// the avatar url for this user.
 	Avatar string `json:"avatar_url" xorm:" varchar(500) 'user_avatar'"`
 
-	// Activate indicates the user is active in the system.
-	Active bool `json:"active" xorm:"user_active"`
```
Comment (6543) [inline server/model/user.go:60]: migration to rm this field?

File: server/api/users.go (hunk 1 above)
Comment (anbraten) [reply server/api/users.go:62]: That's currently nothing as isAdmin is set by WOODPECKER_ADMINS and login, email, avatar are received from the forgea

File: server/model/user.go (hunk 2 above)
Comment (anbraten) [reply server/model/user.go:60]: will add one

Comment (anbraten) [reply server/model/user.go:60]: done

File: server/api/users.go (hunk 1 above)
Comment (lafriks) [reply server/api/users.go:62]: Would be nice if it was possible to set admin option from UI and env variable would be used for initial user creation

Comment (anbraten) [reply server/api/users.go:62]: Yeah definitely. I think we could add some code that sets `isAdmin` at server start for the listed users, but via the UI those setting could be changed later on.

Comment (anbraten) [reply server/api/users.go]: I have adjusted the code and now the admin flag is set to true when a user logs in to the UI.

**Review by lafriks — APPROVED** [review]

File: web/src/components/admin/settings/AdminUsersTab.vue [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+        <InputField :label="$t('admin.settings.users.admin.admin')">
+          <Checkbox
+            :model-value="selectedUser.admin || false"
+            :label="$t('admin.settings.users.admin.placeholder')"
+            @update:model-value="selectedUser!.admin = $event"
+          />
+        </InputField>
+
+        <div class="flex gap-2 justify-center">
```
[the remark this answers is not in the document]
Comment (anbraten) [reply web/src/components/admin/settings/AdminUsersTab.vue:69]: moved it back to left side for now.

**Pull request merged**